### ISSUE TITLE
failed to save outputs: interface conversion: error is *exec.Error, not *exec.ExitError

### ISSUE THREAD

**Issue by hamedhsn** [argoproj/argo-workflows#1207]
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG REPORT

**What happened**:
Seeing this error msg every so often and completely random. It does not happen specifically to a task. I get this for a task and the next run my task works fine. 

The task runs fine and I can see the output but If the next task depends on this task it won't go to the next task.

**What you expected to happen**:
I did not used to see this and start to see that at some point of time when I installed kubeflowpipline and ran a task. However I remove/redeploy argo again but still see the error every so often.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Argo version:
```
$ argo version
argo: v2.2.1
  BuildDate: 2018-10-11T16:25:59Z
  GitCommit: 3b52b26190163d1f72f3aef1a39f9f291378dafb
  GitTreeState: clean
  GitTag: v2.2.1
  GoVersion: go1.10.3
  Compiler: gc
  Platform: darwin/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml
clientVersion:
  buildDate: 2018-07-10T10:13:58Z
  compiler: gc
  gitCommit: 91e7b4fd31fcd3d5f436da26c980becec37ceefe
  gitTreeState: clean
  gitVersion: v1.11.0
  goVersion: go1.10.3
  major: "1"
  minor: "11"
  platform: darwin/amd64
serverVersion:
  buildDate: 2018-12-06T23:13:14Z
  compiler: gc
  gitCommit: 6bad6d9c768dc0864dab48a11653aa53b5a47043
  gitTreeState: clean
  gitVersion: v1.11.5-eks-6bad6d
  goVersion: go1.10.3
  major: "1"
  minor: 11+
  platform: linux/amd64
```

**Other debugging information (if applicable)**:
- workflow result:
```
$ argo get <workflowname>
```
argo get argo-gpu-s3-copy-4qzp8
Name:                argo-gpu-s3-copy-4qzp8
Namespace:           development
ServiceAccount:      argo
Status:              Error
Created:             Sat Feb 02 19:10:05 +0000 (3 minutes ago)
Started:             Sat Feb 02 19:10:05 +0000 (3 minutes ago)
Finished:            Sat Feb 02 19:10:21 +0000 (3 minutes ago)
Duration:            16 seconds
Parameters:
  s3-path:           Shared_data/OULU/small_frames_npy
  local-path:        test2
  bucker-name:       onfido-mlplatform-in
  node-selector:     m4.xlarge

STEP                       PODNAME                            DURATION  MESSAGE
 ⚠ argo-gpu-s3-copy-4qzp8
 └-⚠ list-chunk            argo-gpu-s3-copy-4qzp8-3040831338  16s       failed to save outputs: interface conversion: error is *exec.Error, not *exec.ExitError
- executor logs:
```
$ kubectl logs <failedpodname> -c init
$ kubectl logs <failedpodname> -c wait
```
- workflow-controller logs:
```
$ kubectl logs -n kube-system $(kubectl get pods -l app=workflow-controller -n kube-system -o name)
```

**Comment by wadeholler** [comment]
I just build from master ( plus a unrelated tweak ) and can confirm this behavior - except I get it every run

**Comment by hamedhsn** [comment]
@wadeholler thanks. it fails 80% of times for me. I think we upgraded the k8s cluster version and starts to see this.
Any solution for that?

**Comment by elikatsis** [comment]
@wadeholler there had been a bug on master branch, I fixed it with the PR-1213. Try to delete 'argoexec:latest' from your cluster and build it again using the new dockerfile (if 'argoproj/argoexec:latest' is not updated or if you made any modifications).

**Comment by wadeholler** [comment]
That helped the stated problem above but now submodule support is broken: 

`failed to load artifacts: fatal: No url found for submodule path 'obsfuscated' in .gitmodules`

**Comment by wadeholler** [comment]
my previous reply was for repos that had a submodule reference but no .gitmodules file.  the new argoexec updates that force a submodule update caused this issue.  unrelated to the above.  All is well now. cheers

**Comment by jessesuen** [comment]
I'm pretty sure I had fixed a `exec.Error, not *exec.ExitError` panic conversion as part of the PNS work. Will close this as fixed in v2.3 but please re-open if it is seen again.
